File: FoodieApplication/FavouriteService/src/main/java/com/example/FavouriteService/model/Favourite.java
package com.example.FavouriteService.model;

import lombok.AllArgsConstructor;
import lombok.Data;

import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@Document
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Favourite {

    @Id
    private String favouriteId;
    private String userMailId;
    private List<Restaurant> restaurantList;
    private String isFavourite="white";
//    private String foodName;

}

File: FoodieApplication/frontend/FoodieApplication/src/app/favService/domain/favourite.ts
import { Restaurant } from "src/app/rest_management/domain/restaurant";


export class Favourite {
    favouriteId: string;
    userMailId: string;
    restaurantList: Array<Restaurant>;

    constructor(){
        this.favouriteId='';
        this.userMailId='';
        this.restaurantList=[];
    }
}
